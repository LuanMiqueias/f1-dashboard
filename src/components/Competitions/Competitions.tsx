import React from "react";
import CardCompetition from "../CardCompetition/CardCompetition";
import styles from "./style.module.css";

interface PropsDrives {
  id: number;
  name: string;
  location: {
    country: string;
    city: string;
  };
}
interface IData {
  data: PropsDrives[];
}
const Competitions = ({ data }: IData) => {
  return (
    <div className={styles.container}>
      {data.map((item) => {
        return (
          <CardCompetition
            id={item.id}
            name={item.name}
            location={item.location}
            key={item.id}
          />
        );
      })}
    </div>
  );
};

export default Competitions;
