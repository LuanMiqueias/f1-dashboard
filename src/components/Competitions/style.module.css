.container{
  display: flex;
  align-items: flex-start;
  justify-content: flex-start;
  flex-wrap: wrap;
  gap:8px;
  animation: animaShow .3s linear forwards;
}

@keyframes animaShow{
  0%{
    opacity: 0;
  }
  100%{
    opacity: initial;
  }
}