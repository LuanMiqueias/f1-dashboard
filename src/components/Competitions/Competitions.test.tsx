import { render } from "@testing-library/react";
import Competitions from "./Competitions";

const fakeData = [
  {
    id: 1,
    name: "string",
    location: {
      country: "string",
      city: "string",
    },
  },
];

it("Should math snapshot", () => {
  const { container } = render(<Competitions data={fakeData} />);

  expect(container.firstChild).toMatchSnapshot();
});
