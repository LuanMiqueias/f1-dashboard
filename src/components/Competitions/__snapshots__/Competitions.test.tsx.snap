// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should math snapshot 1`] = `
<div
  class="container"
>
  <div
    class="card_competition"
  >
    <div
      class="content_card_competition"
    >
      <h2>
        string
      </h2>
      <p>
        string
        , 
        string
      </p>
    </div>
  </div>
</div>
`;
